revolv/payments/utils.py

```diff
@@ -1,6 +1,398 @@
+        """
+        new_name = ""
+        for c in name:
+            value = ord(c)
+            if value < 65 or value > 122:
+                new_name += 'z'
+            else:
+                new_name += c
+        return new_name
+
+    def has_payments(self, project):
+        """
+        Checks whether a project has any associated donations, reinvestments, or repayments.
+
+        :args:
+            project: The project to check.
+
+        :return:
+            A boolean of whether the project has associated payment objects.
+        """
+        donations = Payment.objects.donations(project=project, start_date=self.start_date, end_date=self.end_date).count()
+        reinvestments = AdminReinvestment.objects.reinvestments(project=project, start_date=self.start_date, end_date=self.end_date).count()
+        repayments = AdminRepayment.objects.repayments(project=project, start_date=self.start_date, end_date=self.end_date).count()
+
+        return (donations + reinvestments + repayments) != 0
+
+    def move_one_month_forward(self, date):
+        """
+        A helper function that returns a date one month ahead of the date
+        passed in.
+
+        :return:
+            A Python datetime.datetime object with a time value one month ahead of date.
+        """
+        new_month = (date.month)%12 + 1
+        if new_month==1:
+            new_year = date.year + 1
+            new_date = date.replace(month=new_month, year=new_year)
+        else:
+            new_date = date.replace(month=new_month)
+        return new_date
+
+    def set_up_filtered_dict(self):
+        """
+        Sets up the basics of a dictionary containing all necessary information.
+        This sets up:
+        1. cash_in and its various components (repayments, adjustments, donations)
+        2. cash_out and its various components (reinvestments, adjustments)
+        3. date_info and its various components (start_date, end_date, date_list)
+        4. cash_balances
+        5. project_names
+
+        :return:
+            Returns a properly set up dictionary.
+        """
+        filtered_dict = {}
+        filtered_dict['cash_in'] = {'repayments': {}, 'adjustments': {}, 'donations': {}}
+        filtered_dict['cash_out'] = {'reinvestments': {}, 'adjustments': {}}  
+        filtered_dict['date_info'] = {}
+        filtered_dict['cash_balances'] = {'beginning_cash_balance': [], 'change_in_cash': [], 'final_cash_balance': [], 'net_cash_in': [], 'net_cash_out': []}
+        date_info = filtered_dict['date_info']
+        date_info['start_date'] = self.start_date
+        date_info['end_date'] = self.end_date
+        date_info['date_list'] = []
+        filtered_dict['project_names'] = []
+        project_names = filtered_dict['project_names']
+        for proj in self.project_filter:
+            if (self.has_payments(proj)):
+                project_names.append(proj.title)
+
+        for ct in ['cash_in', 'cash_out']:
```

Comment:
how about `cash_type` instead of `ct`? Verbosity is okay if it makes reading the code easier :)